proxy-router/internal/system/validation.go

```diff
@@ -0,0 +1,38 @@
+package system
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type Validator struct {
+}
+
+func NewValidator() *Validator {
+	return &Validator{}
+}
+
+func (v *Validator) ValidateEthResourse(ctx context.Context, url string, chainId *big.Int, timeout time.Duration) error {
```

Comment:
since chainId doesn't change in runtime, it might be convenient to inject it in constructor.